src/test/java/jewellery/inventory/integration/ProductCrudIntegrationTest.java

```diff
@@ -28,6 +28,30 @@
 
 class ProductCrudIntegrationTest extends AuthenticatedIntegrationTestBase {
 
+  @Autowired private UserRepository userRepository;
+  @Autowired private ProductRepository productRepository;
+  @Autowired private ResourceRepository resourceRepository;
+  @Autowired private ResourceInUserRepository resourceInUserRepository;
+  @Autowired private ResourceInProductRepository resourceInProductRepository;
+  private User user;
+  private PreciousStone preciousStone;
+  private User differentUser;
+  private ResourceInUserRequestDto resourceInUserRequestDto;
+  private ResourcesInUserResponseDto resourcesInUserResponseDto;
+  private ProductRequestDto productRequestDto;
+  private ProductRequestDto productRequestDto2;
+  private ProductResponseDto productResponseDto;
+
+  @NotNull
+  private static ResourceInUserRequestDto getResourceInUserRequestDto(
+      User user, PreciousStone preciousStone) {
+    ResourceInUserRequestDto resourceInUserRequestDto = new ResourceInUserRequestDto();
+    resourceInUserRequestDto.setUserId(user.getId());
+    resourceInUserRequestDto.setResourceId(preciousStone.getId());
+    resourceInUserRequestDto.setQuantity(20);
+    return resourceInUserRequestDto;
+  }
```

Comment:
This should be moved down to the other object creation methods.

Reply:
Fixed